### PR TITLE
Add set_mfa_required method

### PR DESCRIPTION
**Pull request by Middledot**
## Summary
Add method that can edit the guild mfa requirements

[docs](https://discord.com/developers/docs/resources/guild#modify-guild-mfa-level)
[discord-api-docs#5072](https://github.com/discord/discord-api-docs/pull/5072)

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

**Review by Lulalaby — DISMISSED** [review]

File: discord/guild.py [hunk 1]
```diff
@@ -1565,15 +1565,36 @@ async def delete(self) -> None:
...
+        -------
         HTTPException
             Deleting the guild failed.
         Forbidden
             You do not have permissions to delete the guild.
         """
-
         await self._state.http.delete_guild(self.id)
 
+    async def set_mfa_required(self, toggle: bool, *, reason: str = None):
```
Comment (Dorukyum) [inline discord/guild.py:1576]: Maybe the `toggle` parameter could be called `required`?

Comment (Lulalaby) [reply discord/guild.py:1576]: I agree

Comment (Dorukyum) [reply discord/guild.py:1576]:
```suggestion
    async def set_mfa_required(self, required: bool, *, reason: str = None) -> None:
```

File: discord/guild.py [hunk 2]
```diff
@@ -1565,15 +1565,36 @@ async def delete(self) -> None:
...
+        """|coro|
+        
+        Set whether it is required to have MFA enabled on your account
+        to perform moderation actions. You must be the guild owner to do this.
+
+        Parameters
+        -----------
+        toggle: :class:`bool`
```
Comment (Dorukyum) [inline discord/guild.py:1584]:
```suggestion
        required: :class:`bool`
```

File: discord/guild.py [hunk 3]
```diff
@@ -1565,15 +1565,36 @@ async def delete(self) -> None:
...
+            Whether it should be required.
```
Comment (Dorukyum) [inline discord/guild.py:1585]:
```suggestion
            Whether MFA should be required to perform moderation actions.
```

File: discord/guild.py [hunk 4]
```diff
@@ -1565,15 +1565,36 @@ async def delete(self) -> None:
...
+            The reason to show up in the audit log.
+
+        Raises
+        -------
+        HTTPException
+            The operation failed.
+        Forbidden
+            You are not the owner of the guild.
+        """
+        await self._state.http.edit_guild_mfa(self.id, toggle, reason=reason)
```
Comment (Dorukyum) [inline discord/guild.py:1596]:
```suggestion
        await self._state.http.edit_guild_mfa(self.id, required, reason=reason)
```

File: discord/http.py [hunk 5]
```diff
@@ -1334,6 +1334,13 @@ def edit_guild(self, guild_id: Snowflake, *, reason: Optional[str] = None, **fie
             reason=reason,
         )
 
+    def edit_guild_mfa(self, guild_id: Snowflake, toggle: bool, *, reason: Optional[str]) -> Response[guild.GuildMFAModify]:
```
Comment (Dorukyum) [inline discord/http.py:1337]:
```suggestion
    def edit_guild_mfa(self, guild_id: Snowflake, required: bool, *, reason: Optional[str]) -> Response[guild.GuildMFAModify]:
```

File: discord/http.py [hunk 6]
```diff
@@ -1334,6 +1334,13 @@ def edit_guild(self, guild_id: Snowflake, *, reason: Optional[str] = None, **fie
...
+        return self.request(
+            Route("POST", "/guilds/{guild_id}/mfa", guild_id=guild_id),
+            json={"level": int(toggle)},
```
Comment (Dorukyum) [inline discord/http.py:1340]:
```suggestion
            json={"level": int(required)},
```

**Review by Dorukyum — CHANGES_REQUESTED** [review]

**Review by Lulalaby — APPROVED** [review]

**Review by Dorukyum — APPROVED** [review]

**Pull request merged**